specs/RuntimeSpec/Actions.md

```diff
@@ -32,35 +43,48 @@ _Requirements:_
 - sets Account `storage_usage` to `account_cost` (genesis config)
 - sets Account `storage_paid_at` to the current block height
 
-NOTE: for the all subsequent actions in the transaction the `signer_id` becomes `receiver_id` until [DeleteAccountAction](#DeleteAccountAction). It allows to execute actions on behalf of the just created account.
-
+**Errors**:
+- If the action tries to create a top level account whose length is no greater than 32 characters, and `predecessor_id` is not
+`registrar_account_id`, the following error will be returned
 ```rust
-pub struct CreateAccountAction {}
+/// A top-level account ID can only be created by registrar.
+CreateAccountOnlyByRegistrar {
+    account_id: AccountId,
+    registrar_account_id: AccountId,
+    predecessor_id: AccountId,
+}
 ```
 
-## DeployContractAction
-
-_Requirements:_
+- If the action tries to create an account that is neither a top-level account or a subaccount of `predecessor_id`,
+the following error will be returned
+```rust
+/// A newly created account must be under a namespace of the creator account
+CreateAccountNotAllowed { account_id: AccountId, predecessor_id: AccountId },
+```
 
-- _`tx.signer_id` to be equal to `receiver_id`_
-- _`tx.public_key` to be `AccessKeyPermission::FullAccess` for the `singer_id`_
+NOTE: for the all subsequent actions in the transaction the `signer_id` becomes `receiver_id` until [DeleteAccountAction](#DeleteAccountAction). It allows to execute actions on behalf of the just created account.
 
-**Outcome**:
-- sets a code for account
+## DeployContractAction
 
 ```rust
 pub struct DeployContractAction {
     pub code: Vec<u8>, // a valid WebAssembly code
 }
 ```
 
-## FunctionCallAction
-
 _Requirements:_
 
-- _`tx.public_key` to be `AccessKeyPermission::FullAccess` or `AccessKeyPermission::FunctionCall`_
+- _`tx.signer_id` to be equal to `receiver_id`_
+- _`tx.public_key` to be `AccessKeyPermission::FullAccess` for the `signer_id`_
 
-Calls a method of a particular contract. See [details](./FunctionCall.md).
+**Outcome**:
+- sets the contract code for account
+
+**Errors**
+
+- If state or storage is corrupted, it may return `StorageError`. Otherwise this action should not lead to errors.
```

Comment:
I don't think spec should talk about storage errors. Until we get to challenges, you shouldn't be able to arrive into inconsistent state

Reply:
I think it is part of the spec because it is possible that we return `StorageError` if state is somehow corrupted.